tests/stdlib/trandom.nim

```diff
@@ -187,3 +187,15 @@ block: # bug #17467
     doAssert x > 1e-4, $(x, i)
       # This used to fail for each i in 0..<26844, i.e. the 1st produced value
       # was predictable and < 1e-4, skewing distributions.
+
+when not defined(js) and not defined(macosx):
+  ## TODO improve `getTime` precision on MacOS
+  block: # bug #17898
+    var unique: HashSet[Rand]
```

Comment:
use a pre-allocated seq instead so that collecting initRand values is as fast as possible otherwise the test is weak and could work "by accident" because `incl` is slow
eg:
```nim
let size = 1000 # better than 10
let vals = newSeq[Rand](size)
for i in 0..<size:
  vals[i] = initRand()

var unique: HashSet[Rand]
proc isUnique[T](a: iterable[T]): bool =
  # xxx in future work, move to std/algorithm
  # implement using HashSet[T]
doAssert isUnique(items(vals))
```